bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -230,6 +195,41 @@ public void initialize() {
         }
     }
 
+    private void poll() {
+        synchronized (jobLock) {
+            try {
```

Comment:
You should add some logic here to check whether or not this handler was disposed while waiting to acquire the jobLock.
```suggestion
        synchronized (jobLock) {
            if (pollingJob == null)  {
                return;
            }
            try {
```

Reply:
No, because this method is finally not called only by the polling job.
It is what I just discovered and why I added this synchronization.

Reply:
This method is not called only by the polling job.
I see what you mean, we should avoid to run it just after the thing handler was disposed.

Reply:
Done